ibm/resource_ibm_is_security_group_target.go

```diff
@@ -0,0 +1,200 @@
+				ValidateFunc: InvokeValidator("ibm_is_security_group_target", isSecurityGroupTargetID),
+			},
+
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Security group name",
+			},
+
+			isSecurityGroupResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Resource Type",
+			},
+		},
+	}
+}
+
+func resourceIBMISSecurityGroupTargetValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 isSecurityGroupTargetID,
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Required:                   true,
+			Regexp:                     `^[-0-9a-z_]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             64})
+
+	ibmISSecurityGroupResourceValidator := ResourceValidator{ResourceName: "ibm_is_security_group_target", Schema: validateSchema}
+	return &ibmISSecurityGroupResourceValidator
+}
+
+func resourceIBMISSecurityGroupTargetCreate(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	securityGroupID := d.Get("security_group_id").(string)
+	targetID := d.Get(isSecurityGroupTargetID).(string)
+
+	createSecurityGroupTargetBindingOptions := &vpcv1.CreateSecurityGroupTargetBindingOptions{}
+	createSecurityGroupTargetBindingOptions.SecurityGroupID = &securityGroupID
+	createSecurityGroupTargetBindingOptions.ID = &targetID
+
+	sg, response, err := sess.CreateSecurityGroupTargetBinding(createSecurityGroupTargetBindingOptions)
+	if err != nil {
+		return fmt.Errorf("Error while creating Security Group Target Binding %s\n%s", err, response)
+	}
+	sgtarget := sg.(*vpcv1.SecurityGroupTargetReference)
+	d.SetId(fmt.Sprintf("%s/%s", securityGroupID, *sgtarget.ID))
+	return resourceIBMISSecurityGroupTargetRead(d, meta)
+}
+
+func resourceIBMISSecurityGroupTargetRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	securityGroupID := parts[0]
+	securityGroupTargetID := parts[1]
+
+	getSecurityGroupTargetOptions := &vpcv1.GetSecurityGroupTargetOptions{
+		SecurityGroupID: &securityGroupID,
+		ID:              &securityGroupTargetID,
+	}
+
+	data, response, err := sess.GetSecurityGroupTarget(getSecurityGroupTargetOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error getting Security Group Target : %s\n%s", err, response)
+	}
+
+	target := data.(*vpcv1.SecurityGroupTargetReference)
+
+	// d.Set(securityGroupTargetID, *group.ID)
+	d.Set("name", *target.Name)
+	d.Set(isSecurityGroupResourceType, *target.ResourceType)
+	d.SetId(fmt.Sprintf("%s/%s", securityGroupID, *target.ID))
```

Comment:
SetId is not required here..

set the other security_group and target arguments